src/path.rs

```diff
@@ -275,6 +275,38 @@ pub fn resolve(raw: impl Into<String>) -> Result<String, StrictPathError> {
     Ok(path)
 }
 
+/// Stores metadata related to the path.
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub struct PathMetadata {
+    is_case_sensitive: bool,
+    store: Option<Store>,
+}
+
+impl PathMetadata {
+    pub fn new(is_case_sensitive: bool, store: Option<Store>) -> Self {
+        PathMetadata {
+            is_case_sensitive,
+            store,
+        }
+    }
+
+    pub fn is_case_sensitive(&self) -> bool {
+        self.is_case_sensitive
+    }
+
+    #[allow(dead_code)]
+    pub fn store(&self) -> Option<Store> {
+        self.store
+    }
```

Comment:
Instead of having methods for these, we can just make the fields `pub`.

Reply:
`is_case_sensitive` is from your code, just to be clear. But I have no issue doing that refactor.

Reply:
Not sure what you mean; this is part of the new `PathMetadata` struct introduced in this PR. Previously, this was just a loose bool in `fn parse_paths`.